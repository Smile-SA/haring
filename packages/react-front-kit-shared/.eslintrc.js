module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
