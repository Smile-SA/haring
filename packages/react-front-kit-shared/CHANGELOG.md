# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
