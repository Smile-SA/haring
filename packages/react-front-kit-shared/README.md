# React Front Kit shared

Shared components and functions.

## Documentation

You can access the documentation here: https://smile-sa.github.io/react-front-kit-doc

## Prerequisite

You must have React 18 or later installed on your project.

## Installation

Install `@smile/react-front-kit-shared` and peer dependencies with:

```bash
npm i @smile/react-front-kit-shared @mantine/core @mantine/hooks @phosphor-icons/react
```

## Styles

Import the styles in your app with:

```js
import '@mantine/core/styles.css';
import '@smile/react-front-kit-shared/style.css';
```

## License

LGPL-3.0
