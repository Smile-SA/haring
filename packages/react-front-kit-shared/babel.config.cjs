const { babelConfig } = require('test');

module.exports = babelConfig;
