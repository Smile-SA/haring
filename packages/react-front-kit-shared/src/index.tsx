/* eslint-disable react-refresh/only-export-components */
'use client';

// component exports
export { ColumnPlus, FolderMove } from './1-styleGuide/Icons';
export { FileIcon } from './1-styleGuide/FileIcon/FileIcon';
export type { IFileIconProps } from './1-styleGuide/FileIcon/FileIcon';
export { Provider } from './3-custom/Provider/Provider';
export type { IProviderProps } from './3-custom/Provider/Provider';
// context exports
export {
  themeContext,
  useMainTheme,
  usePrimaryTheme,
  useSecondaryTheme,
  useThemes,
} from './contexts';
// helper exports
export {
  createThemes,
  isCallback,
  isNotNullNorEmpty,
  typeGuard,
  typeGuardInterface,
} from './helpers';
export type {
  IGuardedType,
  IPrimitiveOrConstructor,
  ITypeGenericInterface,
  ITypeMap,
} from './helpers';
// type exports
export type {
  IAction,
  IActionConfirmModalProps,
  IConfirmAction,
  IConfirmModal,
  IOption,
  IOptions,
  IThemeOverride,
  IThemes,
} from './types';
// type exports
export { mainTheme, primaryTheme, secondaryTheme, themes } from './theme';
