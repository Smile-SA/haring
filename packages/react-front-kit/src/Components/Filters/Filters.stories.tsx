import type { Meta, StoryObj } from '@storybook/react';

import { action } from '@storybook/addon-actions';

import { Filters as Cmp } from './Filters';
import { menu } from './SidebarFilterMenu/SidebarFilterMenu.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/Filters',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const Filters: IStory = {
  args: {
    activeFilters: [
      {
        id: 1,
        label: 'Dupont',
        onRemove: action('remove'),
        value: 'DUPONT',
      },
      {
        id: 1,
        label: 'Martin',
        onRemove: action('remove'),
        value: 'MARTIN',
      },
    ],
    deleteButtonLabel: 'Remove all',
    filterButtonLabel: 'Filtrer',
    sideBarFiltersMenu: menu,
    title: 'Filters actifs',
  },
};
