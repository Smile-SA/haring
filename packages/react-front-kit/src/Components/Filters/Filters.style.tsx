import { createStyles } from '@mantine/styles';

export const useStyles = createStyles((theme) => ({
  activeFilters: {
    display: 'flex',
    flexWrap: 'nowrap',
    marginTop: '16px',
  },
  activeFiltersButtonRoot: {
    minWidth: '100%',
  },
  badgeInner: {
    fontWeight: 600,
    position: 'relative',
    right: '4px',
  },
  badgeRight: {
    position: 'relative',
    right: '4px',
    top: '1px',
  },
  badgeRoot: {
    border: '2px solid white',
    color: 'white',
    cursor: 'pointer',
    fontSize: '12px',
    fontWeight: 'normal',
    marginBottom: '8px',
    marginRight: '8px',
    textTransform: 'capitalize',
  },
  bottom: {
    padding: '24px 24px 48px 24px',
  },
  buttonRemoveRoot: { color: 'white', fontSize: '14px', padding: '0' },
  middle: {
    margin: '0px 24px',
  },
  root: {
    backgroundColor: 'white',
    borderRadius: '16px',
  },
  title: {
    fontWeight: 600,
  },
  top: {
    background: theme.colors.cyan[9],
    border: '8px solid white',
    borderRadius: '16px 16px 0px 0px',
    color: 'white',
    marginBottom: '24px',
    padding: '24px',
  },
}));
