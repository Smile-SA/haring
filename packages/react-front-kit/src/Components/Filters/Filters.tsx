'use client';

import type { IFiltersItem } from './SidebarFilterMenu/SidebarFilterMenu';
import type { ReactElement, ReactNode } from 'react';

import { Badge, Box, Button, Group } from '@mantine/core';
import { TrashSimple, X } from '@phosphor-icons/react';

import { useStyles } from './Filters.style';
import { SidebarFilterMenu } from './SidebarFilterMenu/SidebarFilterMenu';

export interface ISidebarFilter {
  id: number | string;
  label: string;
  onRemove?: (filter: ISidebarFilter) => void;
  value: unknown;
}

export interface IFiltersProps {
  activeFilters?: ISidebarFilter[] | [];
  deleteButtonLabel?: string;
  filterButtonLabel?: string;
  onDeleteButtonClick?: (filters: ISidebarFilter[]) => void;
  onFilterButtonClick?: (filters: ISidebarFilter[]) => void;
  openedMenuIds?: (number | string)[] | undefined;
  sideBarFiltersMenu?: IFiltersItem<number | string>[] | undefined;
  title?: ReactNode;
}

export function Filters(props: IFiltersProps): ReactElement {
  const {
    activeFilters = [],
    title = 'Active filters',
    sideBarFiltersMenu = [],
    onDeleteButtonClick,
    onFilterButtonClick,
    openedMenuIds = undefined,
    deleteButtonLabel = 'Remove all',
    filterButtonLabel = 'Filter',
  } = props;
  const { classes } = useStyles();
  return (
    <Box className={classes.root}>
      <div className={classes.top}>
        <Group position="apart">
          <span className={classes.title}>{title}</span>
          <Button
            className={classes.buttonRemoveRoot}
            leftIcon={<TrashSimple size={12} />}
            onClick={() => onDeleteButtonClick?.(activeFilters)}
            variant="transparent"
          >
            {deleteButtonLabel}
          </Button>
        </Group>
        <div className={classes.activeFilters} />
        {activeFilters.map((filter) => (
          <Badge
            key={filter.id}
            classNames={{
              inner: classes.badgeInner,
              rightSection: classes.badgeRight,
              root: classes.badgeRoot,
            }}
            onClick={() => filter.onRemove?.(filter)}
            pr={3}
            rightSection={<X size={10} />}
            size="xl"
            variant="outline"
          >
            {filter.label}
          </Badge>
        ))}
      </div>
      <div className={classes.middle}>
        <SidebarFilterMenu
          menu={sideBarFiltersMenu}
          openedMenuIds={openedMenuIds}
        />
      </div>
      <div className={classes.bottom}>
        <Button
          classNames={{ root: classes.activeFiltersButtonRoot }}
          color="cyan.9"
          onClick={() => onFilterButtonClick?.(activeFilters)}
          variant="filled"
        >
          {filterButtonLabel} ({activeFilters.length})
        </Button>
      </div>
    </Box>
  );
}
