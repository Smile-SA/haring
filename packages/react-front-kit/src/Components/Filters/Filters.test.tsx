import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { Filters } from './Filters';

describe('Filters', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(<Filters />);
    expect(container).toMatchSnapshot();
  });
});
