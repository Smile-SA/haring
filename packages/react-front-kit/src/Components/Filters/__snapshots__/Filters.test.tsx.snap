// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters matches snapshot 1`] = `
<div>
  <div
    class="mantine-i5l1wq"
  >
    <div
      class="mantine-1f8mro"
    >
      <div
        class="mantine-Group-root mantine-1emc9ft"
      >
        <span
          class="mantine-l9up18"
        >
          Active filters
        </span>
        <button
          class="mantine-UnstyledButton-root mantine-Button-root mantine-1rlikoy"
          data-button="true"
          type="button"
        >
          <div
            class="mantine-1wpc1xj mantine-Button-inner"
          >
            <span
              class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
            >
              <svg
                fill="currentColor"
                height="12"
                viewBox="0 0 256 256"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M216,48H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM192,208H64V64H192ZM80,24a8,8,0,0,1,8-8h80a8,8,0,0,1,0,16H88A8,8,0,0,1,80,24Z"
                />
              </svg>
            </span>
            <span
              class="mantine-1ryt1ht mantine-Button-label"
            >
              Remove all
            </span>
          </div>
        </button>
      </div>
      <div
        class="mantine-bwi36g"
      />
    </div>
    <div
      class="mantine-17q7vz3"
    >
      <div />
    </div>
    <div
      class="mantine-njwlxd"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-s24n6o"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Filter
             (
            0
            )
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
